apps/spark/src/main/java/com/linkedin/openhouse/jobs/util/DataFilesSummary.java

```diff
@@ -0,0 +1,26 @@
+package com.linkedin.openhouse.jobs.util;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.apache.spark.sql.Encoder;
+import org.apache.spark.sql.Encoders;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class DataFilesSummary {
+
+  private Integer content;
+  private Long sumOfFileSizeBytes;
+  private Long totalFileCount;
+
+  /**
+   * Returns the Spark Encoder for this class.
+   *
+   * @return Encoder for ExampleData
+   */
+  public static Encoder<DataFilesSummary> getEncoder() {
+    return Encoders.bean(DataFilesSummary.class);
```

Comment:
how costly is this method -- shall we make Encoder object static and lazily init it only once ?

Reply:
not sure but good call. will share the object to be safe